scripts/irods/test/test_itrim.py

```diff
@@ -78,11 +82,85 @@ def test_itrim_num_copies_repl_num_conflict__issue_3346(self):
             self.admin.assert_icommand('ils -l {0}'.format(filename), 'STDOUT', filename)
 
         finally:
+            self.admin.assert_icommand('ils -l {0}'.format(filename), 'STDOUT', filename) # debugging
             # Clean up.
             self.admin.assert_icommand('irm -f {0}'.format(filename))
             for i in range(replicas):
                 self.admin.assert_icommand('iadmin rmresc resc_{0}'.format(i))
 
+    def test_itrim_option_N_is_deprecated__issue_4860(self):
+        resc1 = 'issue_4860_1'
+        resc2 = 'issue_4860_2'
+        filename = 'test_itrim_option_N_is_deprecated__issue_4860'
+        logical_path = os.path.join(self.admin.session_collection, filename)
+
+        try:
+            lib.create_ufs_resource(self.admin, resc1, test.settings.HOSTNAME_2)
+            lib.create_ufs_resource(self.admin, resc2, test.settings.HOSTNAME_3)
+
+            # Create 3 replicas so that a trim will actually occur.
+            self.admin.assert_icommand(['itouch', logical_path])
+            self.admin.assert_icommand(['irepl', '-R', resc1, logical_path])
+            self.admin.assert_icommand(['irepl', '-R', resc2, logical_path])
+            self.assertTrue(lib.replica_exists_on_resource(self.admin, logical_path, self.admin.default_resource))
+            self.assertTrue(lib.replica_exists_on_resource(self.admin, logical_path, resc1))
+            self.assertTrue(lib.replica_exists_on_resource(self.admin, logical_path, resc2))
+
+            # Ensure the deprecation message appears when a replica is not trimmed.
+            self.admin.assert_icommand(
+                ['itrim', '-N4', logical_path],
+                'STDOUT_MULTILINE',
+                ['Specifying a minimum number of replicas to keep is deprecated.', 'files trimmed = 0'])
+
+            # Ensure the deprecation message appears when a replica is trimmed.
+            self.admin.assert_icommand(
+                ['itrim', '-N2', logical_path],
+                'STDOUT_MULTILINE',
+                ['Specifying a minimum number of replicas to keep is deprecated.', 'files trimmed = 1'])
+
+        finally:
+            self.admin.assert_icommand(['ils', '-l', logical_path], 'STDOUT', filename) # debugging
+            self.admin.run_icommand(['irm', '-f', logical_path])
+            lib.remove_resource(self.admin, resc1)
+            lib.remove_resource(self.admin, resc2)
+
+    def test_itrim_option_age_is_deprecated__issue_7498(self):
+        resc1 = 'issue_7498_1'
+        resc2 = 'issue_7498_2'
+        filename = 'test_itrim_option_age_is_deprecated__issue_7498'
+        logical_path = os.path.join(self.admin.session_collection, filename)
+
+        try:
+            lib.create_ufs_resource(self.admin, resc1, test.settings.HOSTNAME_2)
+            lib.create_ufs_resource(self.admin, resc2, test.settings.HOSTNAME_3)
```

Comment:
The use of `test.settings.HOSTNAME_N` makes sense for topology, but in a zone with one server, are these variables set to resolve to the same host?

I'm looking at the following and wondering how the test succeeds.
https://github.com/irods/irods/blob/e6df49f58759d6609650b82ce4a907d82ee10881/scripts/run_tests.py#L192-L201

Is the testing environment setting those variables?

Reply:
This if statement is only reached when `topology_test` is active:
https://github.com/irods/irods/blob/e6df49f58759d6609650b82ce4a907d82ee10881/scripts/run_tests.py#L189-L201

Reply:
Makes sense. That implies the `HOSTNAME_N` variables are initialized to the same value earlier in the code.

Is that correct?

Reply:
Oops, forgot the rest of the comment...

https://github.com/irods/irods/blob/e6df49f58759d6609650b82ce4a907d82ee10881/scripts/irods/test/settings.py#L6

The values of these hostnames are all the same at time of import as shown above.

Reply:
Excellent. Thanks.